Assets/MirageXR/Tests/NewUI/ContentListView_v2.cs

```diff
@@ -215,12 +215,22 @@ private void OnStepNameChanged(string newTitle)
     {
         _currentStep.instruction.title = newTitle;
         EventManager.NotifyActionModified(_currentStep);
+
+        if (_info.activeInHierarchy)
+        {
+            activityManager.SaveData();
+        }
```

Comment:
could you explain why there is a check?

Reply:
These checks prevent the activity being saved as soon as it is created as we set the names and descriptions when creating the activity. With the checks the user has to go to the correct page and change the name/description themselves for the activity to save

Reply:
I understand, but I think this was relevant to the `onValueChanged` event. Now the `onEndEdit` event is used. But even if `onEndEdit` event is called when text is changed from code, I don't like this check because it doesn't guarantee unforeseen saving. It is better in such a case to unsubscribe the text fields from events, apply the new text and sign the text fields again.